prometheus/src/main/scala/filodb/prometheus/query/PrometheusModel.scala

```diff
@@ -88,15 +106,42 @@ object PrometheusModel {
       Sampl(r.getLong(0) / 1000, r.getDouble(1))
     }.toSeq
 
-    val res = if (samples.isEmpty) None else Some(samples)
+    typ match {
+      case QueryResultType.RangeVectors =>
+        Result(tags,
+          // remove NaN in HTTP results
+          // Known Issue: Until we support NA in our vectors, we may not be able to return NaN as an end-of-time-series
+          // in HTTP raw query results.
+          if (samples.isEmpty) None else Some(samples),
+          None
+        )
+      case QueryResultType.InstantVector =>
+        Result(tags, None, samples.headOption)
+      case QueryResultType.Scalar => ???
```

Comment:
Forgot to replace ??? with impl ?

Reply:
There was never an implementation for Scalar, will add a TODO.